services/provider/server/server.go

```diff
@@ -653,11 +655,21 @@ func (s *OCSProviderServer) GetStorageClassClaimConfig(ctx context.Context, req
 // ReportStatus rpc call to check if a consumer can reach to the provider.
 func (s *OCSProviderServer) ReportStatus(ctx context.Context, req *pb.ReportStatusRequest) (*pb.ReportStatusResponse, error) {
 	// Update the status in storageConsumer CR
-	if err := s.consumerManager.UpdateStatusLastHeatbeat(ctx, req.StorageConsumerUUID); err != nil {
+
+	if _, err := semver.Parse(req.ClientClusterVersion); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Malformed ClientClusterVersion: %v", err)
+	}
+
+	if _, err := semver.Parse(req.ClientOperatorVersion); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Malformed ClientOperatorVersion: %v", err)
+	}
+
+	consStatus := &cs.StatusInfo{ReportStatusRequest: req}
```

Comment:
You should never be able to construct this structure outside of the protobuf package. You are leaking implementation details

Reply:
removed the wrapper, done.